runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileWrapperBuilder.java

```diff
@@ -44,6 +43,9 @@
         ICorfuSMR<T> wrapperObject = (ICorfuSMR<T>) ReflectionUtils.
                             findMatchingConstructor(wrapperClass.getDeclaredConstructors(), args);
 
+        // Close this object since a new one will be instantiated by the VersionLockedObject.
```

Comment:
I think we should clean this up. Please look at #2225 and #2234 , I pushed some fixes and now your tests passes. 

The problem is to create an SMRObject you need to create two objects, one servers as a facade, which is instrumented and passed to the user and the other is managed by the proxy via VersionLockedObject.

What was happening was that the `close` method was being executed on the facade and not the underlying object. Actually, the underlying object which contains the actual state can't be invoked without invoking a facade method that has been annotated as either @Mutator @Accessor @MutatorAccessor, or @PassThrough. 

`close` which ideally isn't a mutator/accessor needs to execute via PassThrough and NOT "not instrumented", but that functionality seems to be broken on master, so i fixed it here #2234 .

Another issue is, the specification for an SMRObject is not well defined. For example, a consumer can access/mutate instrumented code and non-instrumented code, which execute on two different objects (not expected by the user, hence this bug #2225). Based on Java's typing and how our code generation happens, I think the cleanest way to break this circular dependency is to stipulate that the Object definition reserve the default constructor to generate an identity instance (i.e. an empty object that serves as a facade), so line 43 will use that object only as a facade and then the Version locked object can create the other instance with an other constructor/params it requires. Another requirement that we should enforce is that every public method should be annotated, everything else should be private. Right now we can address that by documentation, but later on it would be nice to support it in the annotation processor directly.

Reply:
Adopted. Rather than overloading the semantics of a default constructor, I added a new API call `closeWrapper()` which releases the resources just before the VLO instantiates a new object. But I agree with you -- we need some extensive cleanup in that area.